src/basics.md

```diff
@@ -470,6 +471,84 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Processes a string of text using [`Regex::captures_iter`] to capture multiple
+phone numbers.  The example here is for US convention phone numbers.
+
+```rust, no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use regex::Regex;
+use std::fmt;
+
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+#
+#[derive (PartialEq, PartialOrd, Debug)]
+ enum CountryCode {
+     US,
+ }
+
+#[derive (PartialEq, PartialOrd, Debug)]
+struct PhoneNumber {
+    country: CountryCode,
+    number: String,
+}
```

Comment:
There is no clean way of concatenating the str's  without allocation

The easiest way would be to follow my suggestion of separate &str member for each capture (area, extension, subscriber). Then you don't need to concat and allocate